### PR TITLE
Add support for serializing to a BytesIO buffer

### LINKED ISSUE #69: Support BytesIO object for serialization
Hi GEOLib team,

As a developer I would like GEOLib to support serialization to an in memory BytesIO buffer. At this moment it is only possible to save the generated d-series input files to a file on the local file storage. At VIKTOR, we and our customers (the users of the D-series software) use GEOLib a lot within VIKTOR applications. With the current implementation in GEOLib, workarounds are needed to handle the generated d-series input files in memory and for example serve them as file downloads from the browser or calculate them on a local machine or remote server using VIKTOR. Sinds these applications are hosted on a remote server.

Supporting BytesIO buffers for serialization of the models will solve this problem.

### PR DESCRIPTION
**Pull request by Mbweehuizen**
Hi GEOLib team,

As discussed in issue #69, i would like GEOLib to support BytesIO buffers for serialization. In this PR, I have modified the serialize functions on the models and serializers to both allow local file paths and BytesIO buffers.

These modifications are fully backwards compatible, so the old functionality is unchanged.

Let me know what you think?

closes #69

### CODE REVIEW

**Comment by Mbweehuizen** [comment]
Will add some more tests later

File: geolib/models/serializers.py [hunk 1]
```diff
@@ -10,10 +13,12 @@ class BaseSerializer(BaseDataClass):
...
         return str(self.ds)
 
-    def write(self, filename: FilePath):
-        """Test."""
+    def write(self, filename: Union[FilePath, BytesIO]):
+        """Write serialized model to Filepath or BytesIO buffer"""
+        if isinstance(filename, BytesIO):
+            filename.write(self.render().encode('utf-8'))
         with open(filename, "w") as io:
             io.write(self.render())
```
[the remark this answers is not in the document]
Comment (Mbweehuizen) [reply geolib/models/serializers.py]: Hi Carlos,

You are right, i missed an `else` statement there. I fixed it in the last push.

**Comment by carloslubbers** [comment]
@Mbweehuizen Looks good to me now, do you still want to add some more tests for the other models? If those are present I will merge it for you

**Comment by Mbweehuizen** [comment]
@carloslubbers I will add some more tests for the other models. Incomming soon.

**Comment by Mbweehuizen** [comment]
Tests added.

**Comment by Mbweehuizen** [comment]
@carloslubbers I checking something, please hold merging for now.

**Review by carloslubbers — APPROVED** [review]

**Pull request merged**